internals/src/array_vec.rs

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! A simplified `Copy` version of `arrayvec::ArrayVec`.
+
+use core::fmt;
+
+pub use safety_boundary::ArrayVec;
+
+/// Limits the scope of `unsafe` auditing.
+mod safety_boundary {
+    use core::mem::MaybeUninit;
+    use crate::const_tools::cond_const;
+
+    /// A growable contiguous collection backed by array.
+    #[derive(Copy)]
+    pub struct ArrayVec<T: Copy, const CAP: usize> {
+        len: usize,
+        data: [MaybeUninit<T>; CAP],
+    }
+
+    impl<T: Copy, const CAP: usize> ArrayVec<T, CAP> {
+        // bounds are unstable until 1.61
```

Comment:
This comment is meant to say "bounds are not const until 1.61", right? If so its not necessary IMO, we can see that from the macro call. Same below.

Reply:
Bounds are not allowed in const to be precise. It's useful to explain what specific thing is preventing us to go lower. At least I don't remember all of them.

Reply:
Oh, I get it, I read it wrong. Thanks

Reply:
Sorry to be annoying but can we have `// Trait bounds are unstable until 1.61`. And the one below is definitely wrong `from_raw_parts` is stable in Rust 1.0.0, it became const in 1.64.0

Reply:
Yeah, the correct term is `const`-stable. This PR needs to change anyway because of the condition above.